boa/test/strategies.py

```diff
@@ -99,12 +100,18 @@ def generate_random_string(n):
 
 @_exclude_filter
 def _address_strategy(length: Optional[int] = 100) -> SearchStrategy:
+
+    # TODO: add addresses generated by the env through the env.generate_address() method
+    env_accounts = (
+            {obj.address for obj in boa.env._code_registry.values()}
```

Comment:
it's a bit weird here to introduce a dependency on the global env. should we maybe make it parametrizable? like if `None` is provided then we use `boa.env`, but otherwise a specific env can be given.

Reply:
something like that?
```
@_exclude_filter
def _address_strategy(length: Optional[int] = 100, environment: Optional[EnvType] = None) -> SearchStrategy:

    if environment is None:
        environment = boa.env

    # TODO: add addresses generated by the env through the env.generate_address() method
    env_accounts = (
            {obj.address for obj in environment._code_registry.values()}
    )

    # generate `length` random addresses
    random_strings = generate_random_string(length)
    random_accounts = {format_addr(i) for i in random_strings}

    return _DeferredStrategyRepr(
        lambda: st.sampled_from(list(env_accounts | random_accounts)), "accounts"
    )
```

Then would need to modify all tests to pass a specific env

Reply:
strategies are only used by pytest right? it would make sense to use the environment that is available in the context of the tests.

Reply:
that's true, but you can set up contracts without using the global env